### ISSUE TITLE
string mod function, `code_point_at`, does not return correct code point of lone high surrogate

### ISSUE THREAD

**Issue by jtara1** [boa-dev/boa#1632]
**Describe the bug**
I expected the code point of this malformed utf-16 string, 
`'\uD800'`
to be 
`55296`

node.js 14
```
$ node
Welcome to Node.js v14.17.5.
Type ".help" for more information.
> let s
undefined
> s = '\uD800';
'�'
> s.codePointAt(0)
55296
> s.codePointAt(0).toString(16)
'd800'
```

boa
```
console.log('\uD800'.codePointAt(0));
65533
```

**To Reproduce**
```
// lone high surrogate
console.log('\uD800'.codePointAt(0)); // 55296

// lone low surrogate
console.log('\uDFFF'.codePointAt(0)); // 57343
```

**Build environment (please complete the following information):**

- Rustc version: rustc 1.55.0 (c8dfcfe04 2021-09-06)

**Additional context**
I was working on #894 when I realized something seems wrong with `code_point_at` func. Maybe #736 is relevant. You can assign this issue to me for unit testing.

**Comment by jedel1043** [comment]
Assigned then :)

**Comment by jtara1** [comment]
I'm not sure how we'd detect that it's a malformed unicode encoded string. By the time we encode the JsString as utf-16, it appears to be an acceptable string. I think I'd need to determine if it's malformed from the bytes of the input string.

Rust
```
let c = string.as_str().as_bytes(); // [239, 191, 189]
// same value if we encode string as utf-8
```

**Comment by Razican** [comment]
This was closed in #1659
